tests/ws-dropdown/dropdown-menu.spec.js

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable */
+import {React} from 'imports';
+import {DropdownMenu} from '../../src/ws-dropdown/dropdown-menu';
+
+describe('A DropdownMenu', () => {
+
+  it('filters items', () => {
+    const dm = new DropdownMenu({items: [{label: 't1'}, {label: 't2'}, {label: 't3', stored: true}]});
+    dm.context = {multiple: false};
+
+    // Get all items
+    dm.state.filter = 'irrelevant';
+    expect(dm.getFilteredItems().length).toBe(3);
+    // Get all items with filter string in label and not stored
+    dm.props.filterable = true;
+    dm.state.filter = '1';
+    expect(dm.getFilteredItems().length).toBe(1);
+    dm.state.filter = 't';
+    expect(dm.getFilteredItems().length).toBe(2);
+    // Get all not stored items
+    dm.context.multiple = true;
+    dm.props.filterable = false;
+    dm.state.filter = 'anything';
+    expect(dm.getFilteredItems().length).toBe(2);
+  });
+
+  it('get\'s the item for a selection index', () => {
+    const items = [{label: 'test'}, {label: 'asd'}, {label: 'qwerty'}];
+    const dm = new DropdownMenu({value: [], items});
+    dm.context = {multiple: false};
+
+    expect(dm.getItemAtIndex(0).item).toBe(items[0]);
+    expect(dm.getItemAtIndex(5).item).toBe(items[0]);
+    expect(dm.getItemAtIndex(-1).item).toBe(items[2]);
+
+    dm.state.value = [items[1]];
+    expect(dm.getItemAtIndex(0).item).toBe(items[0]); // index 0 starts in value list
+    expect(dm.getItemAtIndex(1).item).toBe(items[1]); // index 1 starts after value list
+    expect(dm.getItemAtIndex(5).item).toBe(items[0]); // out of bound leads to index 0 which is in value list
+    expect(dm.getItemAtIndex(-1).item).toBe(items[2]);
```

Comment:
To a Fabric newbie such as myself this is not very clear. In JavaScript `-1` usually refers to an index that doesn't exist in the array. Why is it equal to item 2?

This is a place where a short comment would be helpful :).

Reply:
Imagine you have selected the index 0 and press the arrow up key, what would you expect to happen? Probably that the selection jumps to the end of the list.
So index 0 - 1 is out of bounds and will go to last item. Therefore -1 is 2 :)

Reply:
Makes good sense, thanks. I added a comment here.